File: src/haoxiang/Array/MissingNumber.py
__author__ = 'haoxiang'
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        tmp = []
        for i in range(len(nums)+1):
            tmp.append(i)
        print sum(tmp) - sum(nums)

Test = Solution()
test = [0,1,3]
Test.missingNumber(test)

File: src/qiuyi/Array/MissingNumber.cpp
class Solution {
public:
	int missingNumber(vector<int>& nums) {
		int sum1 = 0;
		int sum2 = 0;

		for (auto i = 0; i < nums.size(); i++)
			sum1 += i;
		for (auto num : nums) {
			sum2 += num;
		}

		return sum1 - sum2;
	}
};